script/cibuild

```diff
@@ -74,37 +98,68 @@ main ()
 
         while read scheme
         do
-            build_scheme "$scheme" || status=1
+            # Build each scheme, but stop the entire build if one returns
+            # a nonzero status code
+            build_scheme "$scheme" || status=1 || break
         done
 
         exit $status
     )
 }
 
+#
+# Outputs the first file or directory in pwd that matches the given argument.
+#
 find_pattern ()
 {
     ls -d $1 2>/dev/null | head -n 1
 }
 
-run_xctool ()
+#
+# Invokes xcodebuild using either XCWORKSPACE or XCODEPROJ passing BUILD_OPTIONS
+# and relaying any arguments as additional command-line parameters.
+#
+run_build_tool ()
 {
+    local XCPRETTY_FORMAT_OPTIONS=
+
+    if [ -n "$TRAVIS" ]
+    then
+        # Use a special formatter when running on TravisCI
+        gem install xcpretty-travis-formatter
```

Comment:
Good point. I'm moving it into `.travis.yml`. I had those inline so that they would auto install if someone was testing, but users may need `sudo` if they are using system ruby.